### PR TITLE
Add SharedOperationThrottler to vespalib utils

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review

A cleaned up, de-MessageBus'd version of the implementation that
already exists in the storage module. The old implementation will
be replaced with the vespalib one very soon.

### CODE REVIEW

**Comment by vekterli** [comment]
Existing `SharedOperationHandlerTest` cases will also be moved over as a separate step.

File: vespalib/src/vespa/vespalib/util/shared_operation_throttler.cpp [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+};
+
+/**
+ * Effectively a 1-1 transplant of the MessageBus DynamicThrottlePolicy, but
+ * without an underlying StaticThrottlePolicy and with no need for individual
+ * MessageBus Message/Reply objects.
+ *
+ * Please keep the underlying algorithm in sync with the Java implementation,
+ * as that is considered the source of truth. For descriptions of the various
+ * parameters, also see the Java code.
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/shared_operation_throttler.cpp:37]: Consider adding a link to the Java code.

**Review by geirst — APPROVED** [review]
👍

File: vespalib/src/vespa/vespalib/util/shared_operation_throttler.cpp (hunk 1 above)
Comment (vekterli) [reply vespalib/src/vespa/vespalib/util/shared_operation_throttler.cpp]: Done

**Pull request merged**